### PR TITLE
Adding extra debug logs for the checkpoint logic

### PR DESCRIPTION
**Pull request by Pankaj260100**
Add logging to identify the cause and state when checkpointing occurs in streaming ingestion tasks.

This PR has:

- [x] been self-reviewed.
   - [x] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [ ] added documentation for new or modified features or behaviors.
- [ ] a release note entry in the PR description.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/dev/license.md)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [ ] added integration tests.
- [x] been tested in a test Druid cluster.

### CODE REVIEW

File: indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/SeekableStreamIndexTaskRunner.java [hunk 1]
```diff
@@ -630,6 +630,7 @@ public void run()
           stillReading = !assignment.isEmpty();
 
           SequenceMetadata<PartitionIdType, SequenceOffsetType> sequenceToCheckpoint = null;
+          boolean flagForLogLine = true;
```
Comment (AmatyaAvadhanula) [inline indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/SeekableStreamIndexTaskRunner.java:633]: This is a very vague name.

Also if the goal is to log the segment row and total row values only once when a push is required, you could alternatively store `AppenderatorDriverAddResult pushTriggeringAddResult = null;` outside the loop which is set along with `sequenceToCheckPoint` and you could log the result outside the loop.

**Review by pagrawal10 — APPROVED** [review]
LGTM

**Review by AmatyaAvadhanula — APPROVED** [review]

**Comment by Pankaj260100** [comment]
@AmatyaAvadhanula, can we merge this? We are just adding some logs, and the checks are failing because there are no test cases for new log lines.

**Comment by AmatyaAvadhanula** [comment]
Thanks for the changes, @Pankaj260100

**Pull request merged**